extensions/data-transfer/portability-data-transfer-derived-demo/src/main/java/org/dataportabilityproject/transfer/derived/DerivedDemoImporter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2018 The Data-Portability Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataportabilityproject.transfer.derived;
+
+import org.dataportabilityproject.spi.transfer.provider.ImportResult;
+import org.dataportabilityproject.spi.transfer.provider.Importer;
+import org.dataportabilityproject.transfer.microsoft.spi.types.MicrosoftDerivedData;
+import org.dataportabilityproject.types.transfer.auth.TokenAuthData;
+
+import java.util.UUID;
+
+/** Simulates importing derived data. For demo purposes only! */
+public class DerivedDemoImporter implements Importer<TokenAuthData, MicrosoftDerivedData> {
```

Comment:
if it's not too much work, maybe the data type should be generic and not microsoft, since this is a demo extension. e.g. DemoDerivedData

Reply:
One thing is derived data is specific to the provider and probably won't have any data in common with other providers. We could have a placeholder type but it wouldn't have any data specified in it. The exporting extension would still need to supply a specific type that is understood by the import, so import extensions will still be dependent on the exported type.

Reply:
Placeholder is fine with me, or if you could just add a comment that Microsoft is used since that's the only one we have now, or something like that. Thanks!

Reply:
OK I added a comment in there.